Src/WitsmlExplorer.Api/HttpHandlers/JobHandler.cs

```diff
@@ -101,10 +104,22 @@ public static IResult CancelJob(string jobId, IJobCache jobCache)
             return TypedResults.NotFound();
         }
 
-        [Produces(typeof(IEnumerable<object>))]
-        public static IResult GetReportItems(string jobId, IJobCache jobCache)
+        [Produces(typeof(BaseReport))]
+        public static IResult GetReport(string jobId, IJobCache jobCache, HttpRequest httpRequest, IConfiguration configuration, ICredentialsService credentialsService)
         {
-            return TypedResults.Ok(jobCache.GetReportItems(jobId));
+            EssentialHeaders eh = new(httpRequest);
+            bool useOAuth2 = StringHelpers.ToBoolean(configuration[ConfigConstants.OAuth2Enabled]);
+            string userName = useOAuth2 ? credentialsService.GetClaimFromToken(eh.GetBearerToken(), "upn") : eh.TargetUsername;
+            if (!useOAuth2)
+            {
+                credentialsService.VerifyUserIsLoggedIn(eh, ServerType.Target);
+            }
+            JobInfo job = jobCache.GetJobInfoById(jobId);
+            if (job.Username != userName && (!useOAuth2 || !IsAdminOrDeveloper(eh.GetBearerToken())))
+            {
+                return TypedResults.Forbid();
+            }
+            return TypedResults.Ok(jobCache.GetReportByJobId(jobId));
```

Comment:
You already have the job, so you can just return `TypedResults.Ok(job.Report)`, and remove any changes made in the jobCache.

Reply:
done